internal/clients/http/gcp/gcp_client.go

```diff
@@ -316,6 +316,9 @@ func (c *gcpClient) GetInstanceDescriptionByID(ctx context.Context, id, zone str
 	instanceId := strconv.FormatUint(instance.GetId(), 10)
 	instanceDesc := clients.InstanceDescription{ID: instanceId}
 	for _, n := range instance.NetworkInterfaces {
+		if n.NetworkIP != nil {
```

Comment:
I suggest to even check for n != nil, but I do not know what documentation says.
```suggestion
		if n != nil && n.NetworkIP != nil {
```

Reply:
It is rare to put `nil` into slice/map I would not worry too much about thi.

We have a `ptr.FromOrEmpty` generic helper, can you use that instead?

Reply:
I've used the generic helper, but did not fix the range, I guess it'd be very weird to have nil in an array.

Reply:
Yeah, this was just a suggestion, we do not have to check necessarily:)